packages/transactions/src/signature.ts

```diff
@@ -1,7 +1,19 @@
 import { KeyType } from '@near-js/crypto';
 import { Assignable } from '@near-js/types';
 
+class ED25519Signature extends Assignable { keyType: KeyType = KeyType.ED25519; data: Uint8Array; }
+class SECP256K1Signature extends Assignable { keyType: KeyType = KeyType.SECP256K1; data: Uint8Array; }
+
 export class Signature extends Assignable {
-    keyType: KeyType;
-    data: Uint8Array;
+    ed25519Signature?: ED25519Signature;
+    secp256k1Signature?: SECP256K1Signature;
```

Comment:
The same getter pattern mentioned above would prevent breaking changes here, too.